sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/SubexpressionEliminationSuite.scala

```diff
@@ -154,4 +154,17 @@ class SubexpressionEliminationSuite extends SparkFunSuite {
     equivalence.addExpr(sum)
     assert(equivalence.getAllEquivalentExprs.isEmpty)
   }
+
+  test("Children of CodegenFallback") {
```

Comment:
We does not have a non-leaf expression that use CodegenFallback, except imperative aggregate function, it ends up we can't easily have a end-end test without generated aggregation with imperative function support (WIP).